### PR TITLE
Add Testing to Tictac toe.

### PR DESCRIPTION
**Pull request by adamclasic**
Added tests for the following modules

- Gameplay

- Game-board module

- cleanBoard

- turn

- win

- tie

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required :pray: 
You done an awesome job with the testing. Well done!. However, two of your tests are failing and we need to fix those before moving on.
The failing cases come from the `test/Game.test.js` file. Please fix those and re-submit for another review.


### Martin Afani

File: src/scripts/populateBoard.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (uryelah) [inline src/scripts/populateBoard.js:1]: - [ ] [OPTIONAL] What could you return from here to make it testable? Try to think also what you can test in the function that does not necessarily involve the returned value, the arguments, side effects, etc.

File: test/Game.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { GameBoard } from '../src/scripts/Game';
+
+describe('should check is a valide move', () => {
```
Comment (uryelah) [inline test/Game.test.js:3]: - [ ] [OPTIONAL] Try to improve your descriptions so they really reflect what you are testing inside them. A good idea would be describe the function/module class in the describe first argument.

File: test/Game.test.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  let gameLogic =  GameBoard([null, null, null, null, null, null, null, null, null]);
+  
+  it('should return the board array', () => {
+    expect(gameLogic.board).toStrictEqual([null, null, null, null, null, null, null, null, null]);
```
Comment (uryelah) [inline test/Game.test.js:7]: - [ ] [OPTIONAL] You could also test if the returned value is an array object.

File: test/Game.test.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  });
+  
+});
+
+describe('should check the move is added to board array', () => {
+  let gameLogic = GameBoard([null, null, null, null, null, null, null, null, null]);
+  gameLogic.game(5, gameLogic.board);
+  it('should add X to index 5', () => {
```
Comment (uryelah) [inline test/Game.test.js:15]: - [ ] [OPTIONAL] It's always a good idea to test different test cases to make sure it's not just passing by accident.

File: test/cleanBoard.test.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { cleanBoard } from '../src/scripts/cleanBoard';
+
+describe('should reset the board', () => {
+  const board = cleanBoard([null, null, null, null, "X", null, "O", "X", null]);
+  it('should return an array of 9 nulls', () => {
```
Comment (uryelah) [inline test/cleanBoard.test.js:5]: - [ ] [OPTIONAL] If you want to check if the array has a length of 9 and is filled with null you re checking two things at the same time. You can have two tests them, One for the length, and another for the contents.

File: test/tie.test.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { tie } from '../src/scripts/tie';
+
+describe('should check if ther\'s any room left', () => {
+  it('should return false', () => {
+    expect(tie([null, null, null, null, "X", null, "O", "X", null])).toBeFalsy;
+  });
+
+  it('should return true', () => {
```
Comment (uryelah) [inline test/tie.test.js:8]: - [ ] [OPTIONAL] Always try to test more than one truthy test case.

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+const path = require('path');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist'),
+  },
+  module: {
+    rules: [
```
Comment (uryelah) [inline webpack.config.js:10]: - [ ] Please remove the commented out code before merging the PR.

**Review by uryelah — APPROVED** [review]
Hi **Adam Allalou**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job making the tests pass!
Just make sure you delete the commented out code on the webpack.config file before merging and consider adding more test cases to each of the tested functions. Optionally testing for errors and edge cases would greatly improve your code too!

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being awesome devs and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)
